packages/ember-metal/lib/mixin.js

```diff
@@ -300,29 +296,24 @@ function connectStreamBinding(obj, key, stream) {
 
 function connectBindings(obj, m) {
   // TODO Mixin.apply(instance) should disconnect binding if exists
-  var bindings = m.readableBindings();
-  var key, binding, to;
-  if (bindings) {
-    for (key in bindings) {
-      binding = bindings[key];
-      if (binding) {
-        to = key.slice(0, -7); // strip Binding off end
-        if (isStream(binding)) {
-          connectStreamBinding(obj, to, binding);
-          continue;
-        } else if (binding instanceof Binding) {
-          binding = binding.copy(); // copy prototypes' instance
-          binding.to(to);
-        } else { // binding is string path
-          binding = new Binding(to, binding);
-        }
-        binding.connect(obj);
-        obj[key] = binding;
+  m.forEachBindings((key, binding) => {
+    if (binding) {
+      let to = key.slice(0, -7); // strip Binding off end
```

Comment:
Yes, we do support it for objects.  We do not support it from templates (aka `{{foo-bar blahBinding="asdfadsf"}}` does nothing *magical*).